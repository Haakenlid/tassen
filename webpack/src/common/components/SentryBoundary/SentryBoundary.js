import * as Sentry from '@sentry/browser'

const translation = {
  title: 'Feilmelding',
  subtitle: 'Feilen er registrert. ',
  subtitle2: 'Hvis du vil hjelpe til, skriv noen ord om hva som utløste feilen',
  labelName: 'Navn',
  labelEmail: 'Epost',
  labelComments: 'Hva skjedde?',
  labelClose: 'Lukk',
  labelSubmit: 'Send',
  errorGeneric:
    'Det skjedde en feil (feilception) da jeg skulle sende feilmeldinga. Vennligst prøv en gang til',
  errorFormEntry: 'Noen felter var ikke riktig fylt ut.',
  successMessage: 'Tilbakemeldingen er sendt. Takk skal du ha!',
}

const TraceBack = ({ error }) => (
  <pre
    style={{ whiteSpace: 'pre-wrap', fontSize: '12px' }}
    className="TraceBack"
  >{`${error.stack}`}</pre>
)

class SentryBoundary extends React.Component {
  state = {
    production: process.env.NODE_ENV === 'production',
  }

  componentDidCatch(error, errorInfo) {
    this.setState({ error })
    Sentry.withScope(function(scope) {
      scope.setExtras(errorInfo)
      const eventId = Sentry.captureException(error)
      this.setState({ eventId })
    })
  }

  render() {
    const { error, eventId } = this.state
    if (error) {
      const reportDialog = () =>
        Sentry.showReportDialog({ eventId, ...translation })
      return (
        <div className="SentryBoundary">
          <h1>Søren, heller!</h1>
          {this.production ? <p>Noe gikk galt</p> : <TraceBack error={error} />}
          {eventId && (
            <p>
              Feilen har blitt logget, men du kan også klikke "send melding" for
              å legge inn en manuell feilmelding
              <button onClick={reportDialog}>send melding</button>
            </p>
          )}
          <button onClick={() => this.setState({ error: null, eventId: null })}>
            prøv igjen
          </button>
        </div>
      )
    } else {
      return this.props.children
    }
  }
}

export default SentryBoundary
