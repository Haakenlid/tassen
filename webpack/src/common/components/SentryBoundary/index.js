import SentryBoundary from './SentryBoundary.js'
import './SentryBoundary.scss'
export default SentryBoundary
