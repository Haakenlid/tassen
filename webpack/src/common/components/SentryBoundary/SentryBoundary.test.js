import SentryBoundary from './index'
import renderer from 'react-test-renderer'

describe('SentryBoundary', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <SentryBoundary>
          <div>hello</div>
        </SentryBoundary>,
      )
      .toJSON()
    expect(tree).toEqual(expect.any(Object))
    expect(tree).toMatchSnapshot()
  })
})
