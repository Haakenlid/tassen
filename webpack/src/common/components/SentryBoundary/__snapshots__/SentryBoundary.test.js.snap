// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentryBoundary renders correctly 1`] = `
<div>
  hello
</div>
`;
